src/components/portal/portal.style.js

```diff
@@ -4,7 +4,7 @@ import Portal from "./portal";
 const StyledPortal = styled(Portal)`
   position: absolute;
   top: 0;
-  z-index: 1002;
+  z-index: 3000;
```

Comment:
Shouldnt we use a value defined in `theme` here as well?

Reply:
Why are you even changing this here? The submenus don't use `Portal` so this is unrelated to the bugs you're fixing

Reply:
but without the changes if you will use Menu and Dialog, then Menu will appear over the Dialog background

Reply:
@sjohnson-sage You are completely right! We don't need to set zIndex for whole menu, submenu is enough and because of that I don't need to change portal's zIndex!